crates/router/src/routes/payment_methods.rs

```diff
@@ -251,6 +261,11 @@ pub async fn payment_method_delete_api(
     })
     .into_inner();
 
+    let auth = match auth::is_ephemeral_auth(req.headers()) {
```

Comment:
This is api-key based auth and not ephemeral_auth

Reply:
Should our SDK not handle the delete API? Will the SDK not support delete API or will there be a callback to merchant for this?

Reply:
Yes, for now lets keep retrieve and delete to API-Key itself, for SDK limitations if needed we can add back later.